Switch to jupyterlab

davidrpugh opened the pull request:
PR addressing issue #397. Code is not nearly ready but wanted to open a PR to facilitate discussion.

davidrpugh commented:
Does anyone know how to resize and center align images in markdown? I have been Googling about and have not found a definitive answer. Perhaps this is done in CSS somewhere else in the repo?

davidrpugh commented:
I managed to resize and center align all images by using html in the markdown file for the lesson. Not sure if this is the optimal way or not...

davidrpugh commented:
This PR is ready for some community feedback. Would be particularly interested in getting feedback from maintainers.

souravsingh commented:
Looks good to me. Wdyt @alee @vahtras ?

alee commented on _episodes/01-run-quit.md:
```diff
@@ -5,55 +5,153 @@ exercises: 0
+
+## The JupyterLab Interface
+
+JupyterLab provides flexible building blocks for interactive, exploratory computing. While 
```
Minor nitpick: how about simplifying this paragraph to 

> JupyterLab has many features found in traditional integrated development environments (IDEs) but is focused on providing flexible building blocks for interactive, exploratory computing.

instead?

davidrpugh replied:
Works for me!

alee commented on _episodes/01-run-quit.md:
```diff
@@ -5,55 +5,153 @@ exercises: 0
+and terminals, the command palette, and a list of open tabs in the main work area. A screenshot of 
+the default Left Side Bar is provided below.
+
+<center>
```
We should probably replace all instances of these with `<p align='center'>...</p>` since `<center>` is now obsolete: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/center

See also https://gist.github.com/DavidWells/7d2e0e1bc78f4ac59a123ddf8b74932d

davidrpugh replied:
Thanks for bringing this tom my attention. Didn't realize that `<center>` was obsolete.

alee commented on _episodes/01-run-quit.md:
```diff
@@ -5,55 +5,153 @@ exercises: 0
 ---
-## Python programs are plain text files.
 
-*   They have the `.py` extension to let everyone (including the operating system) 
```
Other maintainer's thoughts on moving the various `Python programs are plain text files` bullet points down to separate sections below? It might still be worthwhile to repeat them up here and below, something like:

> The `.py` extension is a convention not a requirement and it is common to edit Python scripts via an IDE or text editor. However, we will be using Jupyter Lab and Jupyter Notebooks to modify them as Jupyter offers a number of helpful features.

alee commented on _episodes/01-run-quit.md:
```diff
@@ -5,55 +5,153 @@ exercises: 0
+## Starting JupyterLab
+
+### Mac OS X
+To start the JupyterLab server you will need to access the command line through the Terminal. 
```
Should we say Terminal app or other command line terminal application like https://www.iterm2.com/ ? Similarly Windows offers PowerShell or even the [WSL](https://medium.com/hugo-ferreiras-blog/using-windows-subsystem-for-linux-for-data-science-9a8e68d7610c)

davidrpugh replied:
I just pulled this verbiage straight from the `setup.md` file. When I have taught this material the median learner has generally never used a command line terminal application before and thus it seems fine to expose learners to the system defaults. 

Most Windows users, particularly outside US and Western Europe are probably not running Windows 10 (Windows 8 seems to be most widely used here in MENA) so WSL solutions are probably not so useful (unless there are plans to backport?).

davidrpugh commented:
I tried to address all the mandatory changes and as many of the more optional changes as possible. Let me know what you think.

alee commented:
@vahtras and @ntmoore do you have thoughts on making this switch to JupyterLab? I'd like to get full consensus on this before merging since it's a fairly large change. Thanks!

souravsingh commented:
I have no problem switching to JupyterLab.

ntmoore commented:
I am slow to adopt new things so I only use Jupyter (and matplotlib per
that other plotting discussion).  If you think the switch to JupyterLab is
a good idea I will defer to your choice.

On Fri, May 24, 2019 at 4:06 PM A Lee <notifications@github.com> wrote:

> @vahtras <https://github.com/vahtras> and @ntmoore
> <https://github.com/ntmoore> do you have thoughts on making this switch
> to JupyterLab? I'd like to get full consensus on this before merging since
> it's a fairly large change. Thanks!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/swcarpentry/python-novice-gapminder/pull/398?email_source=notifications&email_token=ACDYZOSJE3O33UJFUORXAJDPXBKGVA5CNFSM4HMZOIE2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODWGSBJQ#issuecomment-495788198>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACDYZOT5DYORB5V32F6AIWDPXBKGVANCNFSM4HMZOIEQ>
> .
>


-- 
- - - - - - -   - - - - - - -   - - - - - - -
Nathan Moore
Mississippi River and 44th Parallel
- - - - - - -   - - - - - - -   - - - - - - -

souravsingh commented:
@ntmoore comment made me think, Should we have a section for new users on making a switch from Jupyter notebooks to JupyterLab?

ntmoore commented:
A pointer to this documentation page might be enough,
https://jupyterlab.readthedocs.io/en/stable/user/interface.html

On Sun, May 26, 2019 at 4:48 PM Sourav Singh <notifications@github.com>
wrote:

> @ntmoore <https://github.com/ntmoore> comment made me think, Should we
> have a section for new users on making a switch from Jupyter notebooks to
> JupyterLab?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/swcarpentry/python-novice-gapminder/pull/398?email_source=notifications&email_token=ACDYZORQCBY7YJP642QFWQ3PXMAUVA5CNFSM4HMZOIE2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODWIOC5Q#issuecomment-496034166>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACDYZOUTG75ONLW5GLRUILTPXMAUVANCNFSM4HMZOIEQ>
> .
>


-- 
- - - - - - -   - - - - - - -   - - - - - - -
Nathan Moore
Mississippi River and 44th Parallel
- - - - - - -   - - - - - - -   - - - - - - -

davidrpugh commented:
I cleaned up the discussion of JupyterLab a bit and added a callout box for existing notebook users interested in understanding similarities and differences between JupyterLab and Notebook UIs.

alee commented:
Thanks! this looks good to me - I fixed a few typos and added some minor updates to the text.

davidrpugh commented:
Great! Then I have nothing more to add.  Unless anyone else raises additional issues.

vahtras commented:
It looks to me as the first episode has an illustration of a classical notebook rather than of a notebook in a jupyter lab context - (screen shot of quantum mechanics). Is it possible to update that?

davidrpugh commented:
Sure. It appears that those quantum mechanics notebooks are old (i.e., 5 years+). Here is the link from the episode. 

https://github.com/jrjohansson/qutip-lectures

Can we replace the screenshot with a screenshot of another notebook running in JupyterLab? Suggestions? How about this one from the [JupyterLab docs](https://jupyterlab.readthedocs.io/en/stable/)?

![image](https://user-images.githubusercontent.com/2219562/58453382-92e79600-8123-11e9-9bb3-7fe1e9f9dfe9.png)

davidrpugh commented:
Actually, I think this is a better image (also from the docs)...

![image](https://user-images.githubusercontent.com/2219562/58453694-c7a81d00-8124-11e9-8fb0-a89da9f002da.png)

https://jupyterlab.readthedocs.io/en/stable/user/notebook.html

davidrpugh commented:
I swapped out the screenshots as suggested. If anyone has an idea for better screenshots, then let me know and I can make changes.

vahtras commented:
Nice. I say go ahead

alee commented:
OK - I'll go ahead and merge and we can continue to refine as needed. Thanks @davidrpugh!

Carreau commented on _episodes/01-run-quit.md:
```diff
@@ -102,37 +250,38 @@ keypoints:
-    by entering the Command mode (<kbd>Esc</kbd>/gray) and press the <kbd>Y</kbd> key.
+*   Turn the current cell into a Markdown cell by entering the Command mode (<kbd>Esc</kbd>/gray) 
+    and press the <kbd>M</kbd> key.
+*   `In [ ]:` will disappear to show it is no longer a code cell and you will be able to write in 
```
I believe the `In` part may be gone depending in JupyterLab version, just the `[ ]:` may stay.
